src/stateless/slider.vue

```diff
@@ -0,0 +1,597 @@
+                    :type="inputType"
+                    :size="size"
+                    :disabled="disabled"
+                    :value="currentValue"
+                    :step="step"
+                    :min-value="0"
+                    :max-value="999"
+                    :is-valid="isValidInput"
+                    :theme="theme"
+                    :alignment="alignment"
+                    @input="handleInput"
+                ><span slot="right">%</span></input-element>
+            </div>
+
+            <div class="slider-wrapper" :class="stateClass()"  @mousedown="startDrag" @keydown="keyboard" ref="slider" tabindex="0">
+                <div class="slider-content">
+                    <div class="slider-label">
+
+                        <div class="slider-label__unit" :class="labelClass(0)" ref="min">{{ minLabelValue }}</div>
+ 
+                        <div class="slider-label__unit-tick-container">
+                            <div class="slider-label__unit slider-label__unit-tick"></div>
+                            <div class="slider-label__unit slider-label__unit-tick" :class="tickClass(n)" v-for="n in 18" :key="n">|</div>
+                            <div class="slider-label__unit slider-label__unit-tick"></div>
+                        </div>
+
+                        <div class="slider-label__unit" :class="labelClass(19)" ref="max">{{ maxLabelValue }}</div>
+
+                    </div>
+
+                    <div class="slider">
+                        <div class="bar bar--passive"></div>
+                        <div class="bar bar--active" :class="{'bar--exceeds': limit && currentValue > limitValue}" :style="activeBarStyle"></div>
+                        <div class="bar__handle" :style="handleStyle" ref="handle">
+                            <div class="bar__handle-dot"></div>
+                        </div>
+                    </div>
+                </div>
+
+                <div class="slider-helper-text"></div>
+            </div>
+            
+        </div>
+
+    </div>
+</template>
+
+<script>
+import Input from './input.vue'
+
+export default {
+    components: {
+        inputElement: Input
+    },
+    props: {
+        min: { type: Number },
+        max: { type: Number },
+        limit: { type: Number, required: false },
+        minLabel: { type: String, required: false },
+        maxLabel: { type: String, required: false },
+        label: { type: String },
+        step: { type: Number },
+        value: { type: Number },
+        disabled: { type: Boolean, default: false },
+        disabledText: { type: String, required: false },
+        theme: { type: String, default: 'dark' },
+        size: { type: String, default: 'normal' },
+        alignment: { type: String, default: 'left' },
+        isValid: { type: Function, required: false }
+    },
+    data () {
+        return {
+            domReady: false,
+            index: (this.value - this.min) / this.step,
```

Comment:
What happens if `value` is changed in the parent component. I think it would be better to move this to computed.